src/main/java/org/passay/LengthComplexityRule.java

```diff
@@ -335,69 +329,23 @@ public boolean includes(final Interval i)
      */
     public boolean intersects(final Interval i)
     {
-      return includes(i.getLowerBoundClosed()) || includes(i.getUpperBoundClosed());
-    }
-
-
-    /**
-     * Returns the closed lower bound for this interval.
-     *
-     * @return  closed lower bound
-     */
-    private int getLowerBoundClosed()
-    {
-      return lowerBound.isClosed() ? lowerBound.value : lowerBound.value + 1;
-    }
-
-
-    /**
-     * Returns the closed upper bound for this interval.
-     *
-     * @return  closed upper bound
-     */
-    private int getUpperBoundClosed()
-    {
-      return upperBound.isClosed() ? upperBound.value : upperBound.value - 1;
-    }
-
-
-    @Override
-    public boolean equals(final Object o)
```

Comment:
Since this class is used in a Collection, I'd prefer to keep an `equals` implementation.

Reply:
Yeah I considered adding equals/hashCode to Interval, but since it's used only internally (it should probably be a private inner class no?) and when adding intervals to the map it verifies that they do not intersect (and certainly are not equal), the default Object equals/hashCode is all that is necessary. I can add them if you like anyway, but they don't serve a practical purpose.

Reply:
+1 for making the class private and removing equals/hashCode.

Reply:
ok, added another commit that makes it private, which means all the other unused code can be safely removed as well.